File: spaceship_crm/kotlin/spaceship_crm_kotlin_spring_mustache_jpa/src/main/kotlin/com/spaceship/crm/bean/CustomUserDetails.kt
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.bean

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import com.spaceship.crm.entity.Role
import com.spaceship.crm.entity.User
import java.util.stream.Collectors

class CustomUserDetails(
        var user: User
) : UserDetails {

    override fun getAuthorities(): Collection<GrantedAuthority> {
        return user.roles!!.stream()
                .map {role -> SimpleGrantedAuthority(role.name)}
                .collect(Collectors.toSet())
    }

    override fun getPassword(): String {
        return user.password
    }

    override fun getUsername(): String {
        return user.username
    }

    override fun isAccountNonExpired(): Boolean {
        return true
    }

    override fun isAccountNonLocked(): Boolean {
        return true
    }

    override fun isCredentialsNonExpired(): Boolean {
        return true
    }

    override fun isEnabled(): Boolean {
        return true
    }

}

File: spaceship_crm/scala/spaceship_crm_scala_spring_rest_jpa/src/main/scala/com/spaceship/crm/security/CustomUserDetails.scala
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.security

import java.util

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import com.spaceship.crm.entity.User

import scala.collection.JavaConverters._

class CustomUserDetails(
                         var user: User
                       ) extends UserDetails {



//  .asScala - creates wrapper over collection/array,
// .asJava retrieves the same link
  override def getAuthorities: util.Collection[_ <: GrantedAuthority] =
    user.roles
      .asScala
      .map(role => new SimpleGrantedAuthority(role.name))
      .asJava

  override def getPassword: String = user.password

  override def getUsername: String = user.username

  override def isAccountNonExpired: Boolean = true

  override def isAccountNonLocked: Boolean = true

  override def isCredentialsNonExpired: Boolean = true

  override def isEnabled: Boolean = true
}
